sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -110,16 +113,17 @@ public void pointTo(Object baseObject, long baseOffset, int sizeInBytes) {
     this.baseObject = baseObject;
     this.baseOffset = baseOffset;
     this.sizeInBytes = sizeInBytes;
+    this.headerInBytes = calculateHeaderPortionInBytes(numElements);
```

Comment:
I think we can assign value to `headerInBytes` in constructor instead of `pointTo`